integration/delimited-methods/delimited-methods.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable */
+import { util, configure, Writer, Reader } from 'protobufjs/minimal';
+import * as Long from 'long';
+
+export const protobufPackage = '';
+
+export interface Simple {
+  name: string;
+  age: number;
+  children: string[];
+}
+
+function createBaseSimple(): Simple {
+  return { name: '', age: 0, children: [] };
+}
+
+export const Simple = {
+  encode(message: Simple, writer: Writer = Writer.create()): Writer {
+    if (message.name !== '') {
+      writer.uint32(10).string(message.name);
+    }
+    if (message.age !== 0) {
+      writer.uint32(16).int32(message.age);
+    }
+    for (const v of message.children) {
+      writer.uint32(26).string(v!);
+    }
+    return writer;
+  },
+
+  decode(input: Reader | Uint8Array, length?: number): Simple {
+    const reader = input instanceof Reader ? input : new Reader(input);
+    let end = length === undefined ? reader.len : reader.pos + length;
+    const message = createBaseSimple();
+    while (reader.pos < end) {
+      const tag = reader.uint32();
+      switch (tag >>> 3) {
+        case 1:
+          message.name = reader.string();
+          break;
+        case 2:
+          message.age = reader.int32();
+          break;
+        case 3:
+          message.children.push(reader.string());
+          break;
+        default:
+          reader.skipType(tag & 7);
+          break;
+      }
+    }
+    return message;
+  },
+
+  encodeDelimited(message: Simple, writer: Writer = Writer.create()): Writer {
+    return this.encode(message, writer).ldelim();
```

Comment:
I'm looking at some of the code we create today and it does this:

```
      Nested_InnerMessage.encode(message.message, writer.uint32(18).fork()).ldelim();
```

And the docs for `ldelim` say "Resets to the last state and appends the fork state's current write length as a varint followed by its operations."

This makes me think we need to ensure a fork exists, and change this to be:

```
return this.encode(message, writer.fork()).ldelim();
```

Does that seem right?

Reply:
You're right! It seems like it's the case in the `protobuf-js` implementation [here](https://github.com/protobufjs/protobuf.js/blob/95b56817ef6fb9bdcb14d956c159da49d0889bff/src/type.js#L495).

Reply:
I have a feeling that this should have been caught by a test. Would it be possible to change the tests to write and read 2 messages instead of 1? Then we will really test the main use-case of this feature @AlaaZorkane